config/bgp_cli.py

```diff
@@ -0,0 +1,177 @@
+        table (str): Table name to add new entry to.
+        key (Union[str, Tuple]): Key name in the table.
+        data (Dict): Entry data.
+        create_if_not_exists (bool):
+            In case entry does not exists already a new entry
+            is not created if this flag is set to False and
+            creates a new entry if flag is set to True.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+
+    if not data:
+        raise click.ClickException(f"No field/values to update {key}")
+
+    if create_if_not_exists:
+        cfg[table].setdefault(key, {})
+
+    if key not in cfg[table]:
+        raise click.ClickException(f"{key} does not exist")
+
+    entry_changed = False
+    for attr, value in data.items():
+        if value == cfg[table][key].get(attr):
+            continue
+        entry_changed = True
+        if value is None:
+            cfg[table][key].pop(attr, None)
+        else:
+            cfg[table][key][attr] = value
+
+    if not entry_changed:
+        return
+
+    db.set_entry(table, key, cfg[table][key])
+
+#
+# BGP handlers --------------------------------------------------------------------------------------------------------
+#
+
+def tsa_handler(ctx, db, state):
+    """ Handle config updates for Traffic-Shift-Away (TSA) feature """
+
+    table = CFG_BGP_DEVICE_GLOBAL
+    key = BGP_DEVICE_GLOBAL_KEY
+    data = {
+        "tsa_enabled": state,
+    }
+
+    try:
+        update_entry_validated(db.cfgdb, table, key, data, create_if_not_exists=True)
+        log.log_notice("Configured TSA state: {}".format(to_str(state)))
+    except Exception as e:
+        log.log_error("Failed to configure TSA state: {}".format(str(e)))
+        ctx.fail(str(e))
+
+def wcmp_handler(ctx, db, state):
+    """ Handle config updates for Weighted-Cost Multi-Path (WCMP) feature """
+
+    table = CFG_BGP_DEVICE_GLOBAL
+    key = BGP_DEVICE_GLOBAL_KEY
+    data = {
+        "wcmp_enabled": state,
+    }
+
+    try:
+        update_entry_validated(db.cfgdb, table, key, data, create_if_not_exists=True)
+        log.log_notice("Configured WCMP state: {}".format(to_str(state)))
+    except Exception as e:
+        log.log_error("Failed to configure WCMP state: {}".format(str(e)))
+        ctx.fail(str(e))
+
+#
+# BGP device-global ---------------------------------------------------------------------------------------------------
+#
+
+@click.group(
+    name="device-global",
+    cls=clicommon.AliasedGroup
+)
+def DEVICE_GLOBAL():
+    """ Configure BGP device global state """
+
+    pass
+
+#
+# BGP device-global tsa -----------------------------------------------------------------------------------------------
+#
+
+@DEVICE_GLOBAL.group(
+    name="tsa",
+    cls=clicommon.AliasedGroup
+)
+def DEVICE_GLOBAL_TSA():
+    """ Configure Traffic-Shift-Away (TSA) feature """
+
+    pass
+
+@DEVICE_GLOBAL_TSA.command(
```

Comment:
Do we need it?

Reply:
@stepanblyschak added it like a bonus, since we have a show section